src/v/datalake/proto_to_arrow_scalar.h

```diff
@@ -0,0 +1,138 @@
+#pragma once
+
+#include "datalake/proto_to_arrow_interface.h"
+
+#include <arrow/api.h>
+#include <arrow/array/array_primitive.h>
+#include <arrow/type.h>
+#include <google/protobuf/message.h>
+
+#include <memory>
+#include <stdexcept>
+
+namespace datalake::detail {
+
+template<typename ArrowType>
+class proto_to_arrow_scalar : public proto_to_arrow_interface {
+    using BuilderType = arrow::TypeTraits<ArrowType>::BuilderType;
+
+public:
+    proto_to_arrow_scalar()
+      : _builder(std::make_shared<BuilderType>()) {}
+
+    arrow::Status
```

Comment:
this is interesting, we tend to use std::error-code see the rpc layer. but wondering if the `ok()` is for arrow internals